### DISCUSSION TITLE
How to access and use `persist` value inside x-data ? I get [object Object]

### DISCUSSION THREAD

**Discussion by GeorgeFlorian** [alpinejs/alpine#4068] [1. Help] [answered]
Let's say I have:
```html
<div x-data="{ 
                address: '',
                updateAddress(newAddress) {
                  console.log(newAddress); // logs a string
                  this.address = $persist(newAddress);
                }
              }"
>
```

After updating the address, `x-text="address"` displays `[object Object]`

**Comment by ekwoka** [comment, marked as answer]
This is just using $persist wrong.

$persist is an interceptor and must exist on the context when it initializes.

You need to do

```html
<div x-data="{ 
                address: $persist(''),
                updateAddress(newAddress) {
                  console.log(newAddress); // logs a string
                  this.address = newAddress;
                }
              }"
>
```

during the component initialization the persist is initialized to make address store it's value in local storage.

Was there something about the documentation that made this unclear?

**Reply by GeorgeFlorian** [reply]
Thank you for your fast answer. I got it to work after moving `$persist` to context.

Exactly what you wrote `$persist is an interceptor and must exist on the context when it initializes.` is not in the documentation.
The examples do show `$persist` only used directly in the x-data context, but it doesn't specify it can't or shouldn't be used any other way.

**Reply by ekwoka** [reply]
It is covered slightly in https://alpinejs.dev/plugins/persist#how-it-works but I do guess that isn't quite as explicit as might be needed for someone glancing over it.
